apps/site/lib/site_web/views/trip_plan_view.ex

```diff
@@ -786,4 +795,19 @@ defmodule SiteWeb.TripPlanView do
       _ -> "#{mode_values.mode_name}"
     end
   end
+
+  # Hide monthly pass sections in the case of a Silver Line trip with no transfers from Logain Airport.
+  # Public solely for testing.
+  @spec show_monthly_passes?(Itinerary.t()) :: boolean()
+  def show_monthly_passes?(itinerary), do: !sl_only_trip_from_airport?(itinerary)
+
+  @spec sl_only_trip_from_airport?(Itinerary.t()) :: boolean()
+  defp sl_only_trip_from_airport?(itinerary) do
+    transit_legs = Itinerary.transit_legs(itinerary)
+    first_leg = List.first(transit_legs)
+    route_id = first_leg.mode.route_id
```

Comment:
This will error if the itinerary happens to have no transit legs - please guard against this.

Reply:
Good catch! Updated the implementation and added a test.